### PR TITLE
691: Fixing MUL Portrait Imports

### LINKED ISSUE #691: Loading forces from a MUL file produces incorrect results
### Environment
What version of MekHQ does your issue apply to?

0.43.9-RC3

What operating system are you using?  Windows, Mac, or Linux?

Windows 10

What java version are you using?

Version 8 Update 161, build 1.8.0_161b12

### Description

When you use the "Load forces from a MUL file" feature of the MekHQ, the resulting pilots have some incorrect results, as follows:

1. The gender is random. The MUL files don't store the pilot's gender, even if exported from the MekHQ. When the MUL file is imported, the gender is determined randomly. This can result in some hilarity when you get a male pilot named "Angela" or "Cynthia." :)
2. SPAs (advantages) are lost upon importing.
3. If the campaign is set to assign random portraits, the portraits are chosen randomly even if the pilot already had an assigned portrait.
4. If the pilot had a rank, the name and rank of the pilot does not get set properly when you import the MUL file. The name is saved as "_rank_ _name_" in the MUL file. For example, "Private John Doe." As such, when you import the MUL file, the pilot's name becomes "Private John Doe" in the MekHQ, and his rank is "None."

You can observe all of these behaviors by important the a.mul file in the attached .zip file. See the Files section of this bug report.

### Suggested Fixes

1. Add gender to the MUL files.
2. This one is presumably a bug. Ensure that the advantages get properly imported.
3. If the pilot is already assigned a portrait, do not randomize the portrait.
4. Either save the rank separately in the MUL file, or else enhance the MekHQ to be able to parse the rank from the name. In either case, during the import, have the rank get set properly and exclude the rank from the name.

### Files
[test_mul.zip](https://github.com/MegaMek/mekhq/files/1866009/test_mul.zip)

### PR DESCRIPTION
**Pull request by Windchild292**
This should only be looked at and merged once #1533 has been merged. 

It fixes #691.

For review: I recommend only checking out the current state and ensuring there are no obvious issues I've missed, as I'm preparing a rewrite of the wider utility to fix a few of the bugs in it that are ingrained into the design.

### CODE REVIEW

File: MekHQ/src/mekhq/Utilities.java [hunk 1]
```diff
@@ -883,18 +904,65 @@ else if(u.getEntity() instanceof VTOL) {
...
+    /**
+     * Function that determines what name should be used by a person that is created through crew
+     * And then assigns them a pre-selected portrait, provided one is to their index
+     * Additionally, any extraData parameters should be migrated here
+     * @param p           the person to be renamed, if applicable
+     * @param oldCrew     the crew object they were a part of
+     * @param crewIndex   the index of the person in the crew
+     * @param crewOptions whether or not to run the populateOptionsFromCrew for this person
+     */
+    private static void migrateCrewData(Person p, Crew oldCrew, int crewIndex, boolean crewOptions) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/Utilities.java]: Is this temporary? (given the name `migrate...`)

File: MekHQ/resources/mekhq/resources/CampaignOptionsDialog.properties [hunk 2]
```diff
@@ -92,13 +91,76 @@ useSupportEdgeBox.text=Use Edge for Non-Combat Personnel
...
+useTougherHealing.toolTipText=The healing check will be penalized for every additional hit above 2.
+useTransfers.text=Log Saver - Use Reassign instead of Remove/Assign
+useTimeInService.text=Track Time In Service
+showOriginFaction.text=Show Origin Faction
+randomizeOrigin.text=Randomize Origin
+randomizeDependentsOrigin.text=Randomize Origin for Dependents
+originSearchRadius.text=Random Origin Search Radius
+#family
+FamilyTab.text=Family Options
+minimumMarriageAge.text=Minimum age for marriages:
```
Comment (sixlettervariables) [inline MekHQ/resources/mekhq/resources/CampaignOptionsDialog.properties]: None of this appears to be used in this PR.

File: MekHQ/src/mekhq/Utilities.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/Utilities.java]: It is not, it is used to migrate crew data from MegaMek to MekHQ

File: MekHQ/resources/mekhq/resources/CampaignOptionsDialog.properties (hunk 2 above)
Comment (Windchild292) [reply MekHQ/resources/mekhq/resources/CampaignOptionsDialog.properties]: Part of #1533

**Comment by sixlettervariables** [comment]
Could this be rebased or squashed to make it simpler to review now that #1533 has landed?

**Comment by Windchild292** [comment]
Yeah, this is a Github bug. Give me 15

**Pull request closed without merging**